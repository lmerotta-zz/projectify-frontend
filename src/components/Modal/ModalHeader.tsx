/** @jsxImportSource @emotion/react */
import { ReactNode } from "react";
import * as Styles from "./ModalHeader.styles";

type ModalHeaderProps = {
  children?: ReactNode;
  onClose?: () => void;
};

const ModalHeader = ({ children, onClose }: ModalHeaderProps) => (
  <Styles.ModalHeaderWrapper>
    <Styles.ModalHeaderTitle>{children}</Styles.ModalHeaderTitle>
    {onClose && (
      <Styles.ModalHeaderCloseButtonWrapper
        data-testid="modal-close-button"
        onClick={() => onClose()}
      >
        <Styles.ModalHeaderCloseButton />
      </Styles.ModalHeaderCloseButtonWrapper>
    )}
  </Styles.ModalHeaderWrapper>
);

export default ModalHeader;
