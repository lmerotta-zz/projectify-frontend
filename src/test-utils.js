import React from "react";
import { render } from "@testing-library/react";
import { CacheProvider } from "@emotion/react";
import createCache from "@emotion/cache";

const cache = createCache({ key: "iodajdoiwanoawdni" });
cache.compat = true;

const AllTheProviders = ({ children }) => {
  return <CacheProvider value={cache}>{children}</CacheProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
