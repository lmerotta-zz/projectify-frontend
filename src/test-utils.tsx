import { MockedProvider, MockedProviderProps } from "@apollo/client/testing";
import {
  CircularProgress,
  StyledEngineProvider,
  ThemeProvider,
} from "@mui/material";
import { render as realRender, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode, Suspense } from "react";
import { MemoryRouter, MemoryRouterProps } from "react-router";
import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import theme from "theme";

function renderWithProviders(
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries"> & {
    routerProps?: MemoryRouterProps;
    graphqlProps?: MockedProviderProps;
  }
) {
  const Wrapper = ({ children }: { children?: ReactNode }) => {
    return (
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ToastContainer
            theme="colored"
            autoClose={3000}
            containerId="react-toastify"
          />
          <MockedProvider {...options?.graphqlProps} addTypename={false}>
            <MemoryRouter {...options?.routerProps}>
              <Suspense fallback={<CircularProgress />}>
                <Routes>
                  <Route
                    path="/*"
                    element={
                      <>
                        <h1>TESTSASATATAST</h1>
                        {children}
                      </>
                    }
                  />
                </Routes>
              </Suspense>
            </MemoryRouter>
          </MockedProvider>
        </ThemeProvider>
      </StyledEngineProvider>
    );
  };

  return realRender(ui, { wrapper: Wrapper, ...options });
}

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { renderWithProviders };
