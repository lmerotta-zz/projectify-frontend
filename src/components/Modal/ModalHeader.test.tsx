import { render, fireEvent } from "test-utils";
import ModalHeader from "./ModalHeader";

describe("ModalHeader unit tests", () => {
  it("Does not render a close button", () => {
    const result = render(<ModalHeader>This is my title</ModalHeader>);

    expect(() => result.getByTestId("modal-close-button")).toThrow();
    expect(result.asFragment()).toMatchSnapshot();
  });

  it("Renders a close button that reacts on click", () => {
    const close = jest.fn();
    const result = render(
      <ModalHeader onClose={close}>This is my title</ModalHeader>
    );

    expect(result.getByTestId("modal-close-button")).toBeVisible();
    expect(result.asFragment()).toMatchSnapshot();

    fireEvent.click(result.getByTestId("modal-close-button"));

    expect(close).toHaveBeenCalled();
  });
});
