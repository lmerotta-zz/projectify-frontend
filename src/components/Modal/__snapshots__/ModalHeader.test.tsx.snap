// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader unit tests Does not render a close button 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
}

.emotion-1 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-weight: 100;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
    class="emotion-0"
  >
    <h5
      class="emotion-1"
    >
      This is my title
    </h5>
  </div>
</DocumentFragment>
`;

exports[`ModalHeader unit tests Renders a close button that reacts on click 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom-width: 2px;
  --tw-border-opacity: 1;
  border-color: rgba(229, 231, 235, var(--tw-border-opacity));
}

.emotion-1 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-weight: 100;
  --tw-text-opacity: 1;
  color: rgba(39, 39, 42, var(--tw-text-opacity));
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  position: relative;
  top: 0.375rem;
}

.emotion-3 {
  width: 1rem;
  height: 1rem;
}

<div
    class="emotion-0"
  >
    <h5
      class="emotion-1"
    >
      This is my title
    </h5>
    <div
      class="emotion-2"
      data-testid="modal-close-button"
    >
      <svg
        class="emotion-3"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
